Update syno-plex-update.sh

programatix opened the pull request:
Retrieve the "Server Update Channel" from the preference file and decide which version to download.

programatix committed:
Update syno-plex-update.sh

Retrieve the "Server Update Channel" from the preference file and decide which version to download.

programatix committed:
Update syno-plex-update.sh

Update the in-script comment.

YuriyGuts commented on syno-plex-update.sh:
```diff
@@ -89,6 +93,14 @@ function download_release_metadata {
     local response
+    
+    is_public_channel=$(is_public_channel)
+    if [ "$is_public_channel" -eq "0" ]; then
+        write_log 'Public channel'
```
My preferences file (free tier) doesn't have this substring and this line raises an error:
```
line 98: [: : integer expression expected
```

I think the function needs something like a `grep -oP ... || echo "0"` to handle the unhappy case.

programatix committed:
Update syno-plex-update.sh

Added the fall-through statement in case of missing parameter as suggested. Tested and working.

programatix commented:
Just in case if you're wondering how it looks like in Plex UI setting.
![image](https://user-images.githubusercontent.com/3293865/144518405-1055c332-d0dd-43f0-95c6-c5e5f272a330.png)

The `Server update Channel` option doesn't do anything special except being used when checking for available update by the server.

YuriyGuts reviewed (approved):
Thanks 👍
